storage/bucket.go

```diff
@@ -854,6 +888,62 @@ func (b *BucketAttrs) toRawBucket() *raw.Bucket {
 	}
 }
 
+func (b *BucketAttrs) toProtoBucket() *storagepb.Bucket {
+	// Copy label map.
+	var labels map[string]string
+	if len(b.Labels) > 0 {
+		labels = make(map[string]string, len(b.Labels))
+		for k, v := range b.Labels {
+			labels[k] = v
+		}
+	}
+
+	// Ignore VersioningEnabled if it is false. This is OK because
+	// we only call this method when creating a bucket, and by default
+	// new buckets have versioning off.
+	var v *storagepb.Bucket_Versioning
```

Comment:
All of this repeated code makes me wish we had some common setter interface that was satisfied by both the proto object and the apiary object

Reply:
Yeah, it's a bummer 🤷🏻  wish I didn't have to write all of this either :)